specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,538 @@
+ 
+Technical calls:
+
+1. `alice` calls `games::mt_transfer({"receiver_id": "bob", "amount": "500000000", "token_id": "g133"})`.
+
+#### Simple batch transfer
+
+Alice wants to send and a 1 unique(nft) gemstone, 5 gold and 10 silver tokens to Bob.
+
+Let's assume the following:
+- The unique nft `gem` token is represented by `games` with token_id
+`uu2`
+- The `gold` token is represented by the `games` with token_id `g133`.
+- The `silver` token is represented by the `games` with token_id `s133`.
+- Alice's account is `alice`.
+- Bob's account is `bob`.
+- The precision for `gold` on the games contract is `10^8`. 
+- The precision for `silver` on the games contract is also `10^8`. 
+- The 5 gold tokens is `5 * 10^8` or as a number is `500000000`. 
+- The 10 silver tokens is `10 * 10^8` or as a number is `1000000000`. 
+- The  1 gem token is `1` or as a number is `1`
+
+High-level explanation:
+
+Alice needs to issue one transaction to `games` contract to transfer 5 gold tokens and 10 silver tokens (multiplied by precision) and 1 gem to Bob.
+ 
+Technical calls:
+
+1. `alice` calls `games::mt_transfer_batch({"receiver_id": "bob", "amounts": ["500000000", "1000000000", "1"], "token_ids": ["g133", "s133", "uu2"]})`.
+
+
+
+#### Token deposit to a contract 
+
+Alice wants to deposit `gold` tokens to a compound interest contract to earn some rewards.
+
+Let's assume the following:
+- The `gold` token is represented  by the `games` contract with token_id `g133` .
+- Alice's account is `alice`.
+- The compound interest contract is `compound`.
+- The precision on `gold` token is `10^18`. 
+- The 1000 tokens is `1000 * 10^18` or as a number is `1000000000000000000000`. 
+- The compound contract can work with many different token contracts and types.
+
+High-level explanation:
+
+Alice needs to issue a single transaction to `games` that will internally issue a cross contract call to `compound`. 
+
+The initial transaction to `games` is made with `compound` as the receiver of a set token_ids and amounts from `alice`.
+
+This call then waits on a response from `compound`. If `compound` responds with failure, the tx is aborted. 
+
+Otherwise `games` contract accepts the results and resolves the promise completing the transaction.
+
+- If transfer succeeded, `compound` can increase local ownership for `alice` to 1000 for `gold` token_id `g133`
+
+- If transfer fails, `compound` doesn't need to do anything in current example, but maybe can notify `alice` of unsuccessful transfer.
+
+Technical calls:
+1. `alice` calls `games::mt_transfer_call({"receiver_id": "compound", amount: "1000000000000000000000", "token_id": "g133", msg: "interest-building"})`.
+   During the `mt_transfer_call` call, `compound` does the following:
+     fn mt_on_transfer(
+        &mut self,
+        sender_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        msg: String,
```

Comment:
to support binary, I suggest using Base64 format
```suggestion
       msg: Base64VecU8,
```

Reply:
Hey we were just talking about this and wanted to know if you had a particular use case for this being base64VecU8?

Reply:
we talked about this offline and string seems to be sufficient here.